src/checks/mapChecks.ts

```diff
@@ -0,0 +1,25 @@
+import {AslChecker, StateMachineErrorCode} from "../types";
+import {checkPropertyWorkflow} from "./checkProperty";
+
+export const mapChecks: AslChecker = (definition) => {
+    return [
+        ...checkPropertyWorkflow({
+            definition,
+            filter: ({state}) => {
+                return state.Type === 'Map'
+            },
+            propCheck: 'exactlyOne',
+            props: ["ItemProcessor", "Iterator"],
+            errorCode: StateMachineErrorCode.MapItemProcessorError,
+        }),
```

Comment:
easy enough to add these new checks but I dislike how each one is a separate traversal

Some of the checkers need to traverse the full graph but these simple property checks could be done in a single traversal and array of checkers. that's an improvement for another time.